python/src/openturns/PythonWrappingFunctions.hxx

```diff
@@ -978,6 +978,125 @@ convert< Indices, _PySequence_ >(Indices inP)
   return point;
 }
 
+template <>
+struct traitsPythonType< IndicesFrozenCollection >
+{
+  typedef _PySequence_ Type;
+};
+
+template <>
+inline
+IndicesFrozenCollection
+convert< _PySequence_, IndicesFrozenCollection >(PyObject * pyObj)
+{
+  // Check whether pyObj follows the buffer protocol
+  if (PyObject_CheckBuffer(pyObj))
+  {
+    Py_buffer view;
+    if (PyObject_GetBuffer(pyObj, &view, PyBUF_FORMAT | PyBUF_ND | PyBUF_ANY_CONTIGUOUS) >= 0)
+    {
+      if (view.ndim == 2 &&
+          view.itemsize == traitsPythonType<UnsignedInteger>::buf_itemsize &&
+          view.format != NULL &&
+          strcmp(view.format, pyBuf_formats[traitsPythonType<UnsignedInteger>::buf_format_idx]) == 0)
+      {
+        const UnsignedInteger* data = static_cast<const UnsignedInteger*>(view.buf);
+        const UnsignedInteger size = view.shape[0];
+        const UnsignedInteger dimension = view.shape[1];
+        IndicesFrozenCollection indices( size, dimension );
+        if (PyBuffer_IsContiguous(&view, 'C'))
+        {
+          // 2-d contiguous array in C notation, we can directly copy memory chunk
+          std::copy(data, data + size * dimension, &indices(0,0));
+        }
+        else
+        {
+          for (UnsignedInteger j = 0; j < dimension; ++j)
+            for(UnsignedInteger i = 0; i < size; ++i, ++data)
+              indices(i, j) = *data;
+        }
+        PyBuffer_Release(&view);
+        return indices;
+      }
+      PyBuffer_Release(&view);
+    }
+    else
+      PyErr_Clear();
+  }
+
+  // use the same conversion function for numpy array/matrix, knowing numpy matrix is not a sequence
+  if ( PyObject_HasAttrString(pyObj, const_cast<char *>("shape")) )
+  {
+    ScopedPyObjectPointer shapeObj(PyObject_GetAttrString( pyObj, "shape" ));
+    if ( !shapeObj.get() ) throw;
+
+    Indices shape( checkAndConvert< _PySequence_, Indices >( shapeObj.get() ) );
+    if ( shape.getSize() == 2 )
+    {
+      UnsignedInteger size = shape[0];
+      UnsignedInteger dimension = shape[1];
+      ScopedPyObjectPointer askObj(PyTuple_New(2));
+      ScopedPyObjectPointer methodObj(convert< String, _PyString_ >("__getitem__"));
+      IndicesFrozenCollection indices( size, dimension );
+      try
+      {
+        for ( UnsignedInteger i = 0; i < size; ++ i )
+        {
+          PyTuple_SetItem( askObj.get(), 0, convert< UnsignedInteger, _PyInt_ >(i) );
+          for ( UnsignedInteger j = 0; j < dimension; ++ j )
+          {
+            PyTuple_SetItem( askObj.get(), 1, convert< UnsignedInteger, _PyInt_ >(j) );
+            ScopedPyObjectPointer elt(PyObject_CallMethodObjArgs( pyObj, methodObj.get(), askObj.get(), NULL));
+            if (elt.get())
+            {
+              indices( i, j ) = checkAndConvert<_PyInt_, UnsignedInteger>(elt.get());
+            }
+          }
+        }
+      }
+      catch (InvalidArgumentException &)
+      {
+        throw;
+      }
```

Comment:
The catch emits the same exception here, what did you have in mind ?

Reply:
Copy/paste mistake; this was needed for MatrixImplementation and the like because a pointer had to be deleted, but this is not the case here, will fix it.